app/misc/images.ts

```diff
@@ -1,40 +1,74 @@
+let gh = require('octonode');
+
+// local cache for profile picture URLs
 let images = {};
-// let imageFiles = ["dog1.jpg", "dog2.jpg", "dog3.jpg", "dog4.jpg", "dog5.jpg"];
-let imageFiles = ["jarjar.jpg", "yoda.png", "obiwan.jpg"];
-let imageCount = 0;
-let githubAvatarUrl = require('github-avatar-url');
+
+/**
+ * Helper function for getting the letter icon for a given author.
+ * 
+ * @param name Name of the commit's author
+ * @returns filepath for the picture containing the author's first initial
+ */
+function getLetterIcon(name: string) {
+  let first = name.trim().charAt(0).toUpperCase();
+  return "node_modules/material-letter-icons/dist/png/" + first + ".png";
+}
 
 function getName(author: string) {
   let name = author.split("<")[0];
   return name;
 }
 
-function img4User(name:string) {
-  let pic;
-  let first = name.trim().charAt(0).toUpperCase();
-  pic =  "node_modules/material-letter-icons/dist/png/" + first + ".png";
-  return pic;
+// No longer in use, but keeping it here in case of any potential 
+// backward-compatibility issues on other VisualGit branches.
+function img4User(name:string) {  
+  return getLetterIcon(name);
 }
 
+/** Retreives the URL for the given author's GitHub profile picture.*/
 function imageForUser(name: string, email: string, callback) {
-  let pic;
-  githubAvatarUrl(name, function(err,avatarURL){
-    if(!err) {
-      console.log("avatar url: avatarURL");
-      pic = avatarURL;
-    } else {
-      let first = name.trim().charAt(0).toUpperCase();
-      pic =  "node_modules/material-letter-icons/dist/png/" + first + ".png";
+  
+  let pic;  
+  
+  if (email.includes('@users.noreply.github.com')) {
+    
+    // Extract GitHub username from noreply email (<number>+<username>@users.noreply.github.com)
+    // This won't work for users that are NOT hiding their email in the GitHub settings.
+    // See https://help.github.com/en/articles/about-commit-email-addresses for more info.
+    let username = email.replace('@users.noreply.github.com','').replace(/^(\d){4,}\+/,'');
```

Comment:
You can use account.login(initialized in `authenticate.ts`) to get the usersname.